File: docker-envs/Dockerfile.java
FROM gradle:4.10.2-jdk8

USER root

RUN apt-get update && apt-get install --upgrade dnsutils python-pip -y
RUN pip install requests PyYAML

COPY tests/env_java_tests/spark-app/ /apps/tests/env_java_tests/spark-app/
WORKDIR /apps/tests/

# install dependencies
RUN cd env_java_tests/spark-app/ && sed -ie 's/id "com\.github\.johnrengelman\.shadow".*//' build.gradle && \
    gradle classes

COPY . /apps/

EXPOSE 15003

CMD cd env_java_tests/spark-app/ && gradle run

File: docker-envs/Dockerfile.php
FROM php:7.2.10-apache

RUN apt-get update && apt-get install --upgrade dnsutils python-pip -y
RUN pip install requests PyYAML

RUN sed -i '0,/Listen [0-9]*/s//Listen 15002/' /etc/apache2/ports.conf

RUN mkdir /var/www/html/lib/ && cd /var/www/html/lib && \
    curl -sL 'https://github.com/smarty-php/smarty/archive/v3.1.32.tar.gz' | tar xzf - && \
    curl -sL 'https://github.com/twigphp/Twig/archive/v1.20.0.tar.gz' | tar xzf - && \
    curl -sL 'https://github.com/twigphp/Twig/archive/v1.19.0.tar.gz' | tar xzf -

COPY  . /apps/
COPY tests/env_php_tests/* /var/www/html/

WORKDIR /apps/tests/
